packages/entities/src/lib/entity.query.ts

```diff
@@ -108,3 +111,24 @@ export function getEntity(
 
   return pluck(entity);
 }
+
+export function selectEntityByPredicate<
+  R extends getEntityType<S, Ref>[],
+  K extends keyof getEntityType<S, Ref>,
+  S extends EntitiesState<Ref>,
+  Ref extends EntitiesRef = DefaultEntitiesRef
+>(
+  predicate: ItemPredicate<getEntityType<S, Ref>>,
+  options?: {
+    pluck?: K | ((entity: getEntityType<S, Ref>) => R);
+  } & BaseEntityOptions<Ref>
+): OperatorFunction<S, getEntityType<S, Ref> | undefined> {
+  const { ref: { entitiesKey } = defaultEntitiesRef, pluck } = options || {};
+
+  return pipe(
+    select<S, Ref>((state) =>
+      findEntityByPredicate(state[entitiesKey], predicate)
```

Comment:
Do it only once in the outer scope and query the id:

```ts
const entity = findEntityByPredicate(state[entitiesKey], predicate)
return entitiesChanged, map(checkPluck(entity, pluck), distinctUntilChanged());
```

Reply:
Do you mean this?

```ts
const { ref = defaultEntitiesRef, pluck } = options || {};

  return pipe(
    select<S, Ref>((state) => findEntityByPredicate(state, ref, predicate)),
    map((entity) => checkPluck(entity, pluck))
  );
```